src/components/Frequency/Frequency.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useState, useMemo, useCallback, useRef } from "react";
+import { nanoid } from "nanoid";
+import Field from "../internal/Field";
+import Grid from "../Grid";
+import InternalInput from "../internal/InternalInput";
+import InternalRadioGroup from "../internal/InternalRadioGroup";
+import InternalSelect from "../internal/InternalSelect";
+import { InternalFrequencyProps } from "./types";
+import { useMergedProps } from "../../hooks/useMergedProps";
+import { defaultFrequencyProps } from "./defaultFrequencyProps";
+import mergeRefs from "../../utils/mergeRefs";
+import { ALL_FREQUENCY_OPTIONS, FrequencyValue } from "../../values";
+
+function Frequency(props: InternalFrequencyProps) {
+  const mergedProps = useMergedProps(props, defaultFrequencyProps);
+  const {
+    mode,
+    label,
+    annually,
+    innerRef = null,
+    error,
+    quarterly,
+    monthly,
+    fortnightly,
+    weekly,
+    amountPrefix,
+    amountPlaceholder,
+    selectPlaceholder,
+    amountMaxLength,
+    helpText,
+    disabled,
+    optional,
+    testId,
+    onFocus,
+    onBlur,
+    onChange,
+    onMouseDown,
+  } = mergedProps;
+  const [labelId] = useState(() => `frequency-label-${nanoid()}`);
+  const [auxId] = useState(() => `frequency-aux-${nanoid()}`);
+
+  const frequencyOptions = useMemo(() => {
+    const frequencyPropsMap = {
+      annually,
+      quarterly,
+      monthly,
+      fortnightly,
+      weekly,
+    };
+    return ALL_FREQUENCY_OPTIONS.filter(
+      (option) => frequencyPropsMap[option.value]
+    );
+  }, [annually, quarterly, monthly, fortnightly, weekly]);
+
+  const [internalValue, setInternalValue] = useState<FrequencyValue>({
+    amount: "",
+    frequency: "",
+  });
+
+  const value = props.value ?? internalValue;
+
+  const onFrequencyChange: React.ChangeEventHandler<
+    HTMLInputElement | HTMLSelectElement
+  > = (event) => {
+    const { name: key, value: updatedValue } = event.target;
+    if (key !== "amount" && key !== "frequency") {
+      return;
+    }
+    onChange?.({
```

Comment:
nice

Reply:
for readability, wouldn't
```javascript
if(onChange) {
    onChange()
}
```
make more sense?

Reply:
I don't mind either one. I am fine reading the optional param but if you prefer an if statement I am fine with that